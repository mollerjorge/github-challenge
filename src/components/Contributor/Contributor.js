import React from "react";
import Proptypes from "prop-types";
import ContributorHolder from "./Contributor.style.js";

const Contributor = ({ children: { login, avatar_url, contributions } }) => {
	return (
		<ContributorHolder>
			<img width="50px" src={avatar_url} alt="user avatar" />
			<div style={{ marginLeft: "2rem" }}>
				<h3> {login}</h3>
				<span>Contributions: {contributions}</span>
			</div>
		</ContributorHolder>
	);
};

Contributor.propTypes = {
	children: Proptypes.shape({
		login: Proptypes.string,
		avatar_url: Proptypes.string,
		contributions: Proptypes.number
	})
};

export default Contributor;
