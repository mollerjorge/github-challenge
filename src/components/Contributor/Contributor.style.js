import styled from 'styled-components';

const ContributorHolder = styled.div`

	display: flex;
	height: 5rem;
	margin-bottom: 1rem;
	align-items: center;
	img {
		width: 5rem;
		height: 5rem;
	}

	

`;

export default ContributorHolder;