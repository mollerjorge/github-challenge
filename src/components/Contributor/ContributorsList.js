import React from "react";
import Contributor from "./Contributor";
import { List } from "react-virtualized";
import 'react-virtualized/styles.css'

class ContributorsList extends React.Component {
	rowRenderer = ({
		key, // Unique key within array of rows
		index, // Index of row within collection
		isScrolling, // The List is currently being scrolled
		isVisible, // This row is visible within the List (eg it is not an overscanned row)
		style // Style object to be applied to row (to position it)
	}) => {
		const content = <Contributor key={this.props.contributors[index].id}>
				{this.props.contributors[index]}
			</Contributor>;
		return (
			<div key={key} style={style}>
				{content}
			</div>
			
		);
	
	};

	render() {
		return (
			<React.Fragment>
				<List
					width={300}
					height={500}
					rowCount={this.props.contributors.length}
					rowHeight={100}
					rowRenderer={this.rowRenderer}
				/>
				
			</React.Fragment>
		);
	}
}

export default ContributorsList;
