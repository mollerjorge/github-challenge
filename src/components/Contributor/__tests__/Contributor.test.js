import {shallow} from 'enzyme';
import React from 'react';
import Contributor from '../Contributor';

describe('Contributor', () => {
	const mockChildren = {
		login: "george",
		avatar_url: 'http://mockavatarurl.com',
		contributions: 4
	}

	const contributor = shallow(<Contributor children={mockChildren} />);


	it('renders correctly', () => {
		expect(contributor).toMatchSnapshot();
	});

	it('displays the avatar_url correctly', () => {
		expect(contributor.find('img').prop('src')).toEqual('http://mockavatarurl.com')
	});
	it('displays the login name correctly', () => {
		expect(contributor.find('h3').text().trim()).toEqual('george')
	});
	it('displays the contributions correctly', () => {
		expect(contributor.find('span').text()).toEqual('Contributions: 4')
	});
});