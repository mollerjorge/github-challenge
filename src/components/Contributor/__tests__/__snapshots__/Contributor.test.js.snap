// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contributor renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Contributor>
    Object {
        "avatar_url": "http://mockavatarurl.com",
        "contributions": 4,
        "login": "george",
      }
</Contributor>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <img
          alt="user avatar"
          src="http://mockavatarurl.com"
          width="50px"
/>,
        <div
          style={
                    Object {
                              "marginLeft": "2rem",
                            }
          }
>
          <h3>
                     
                    george
          </h3>
          <span>
                    Contributions: 
                    4
          </span>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "user avatar",
          "src": "http://mockavatarurl.com",
          "width": "50px",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
               
              george
</h3>,
            <span>
              Contributions: 
              4
</span>,
          ],
          "style": Object {
            "marginLeft": "2rem",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "george",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "george",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Contributions: ",
                4,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Contributions: ",
              4,
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <img
            alt="user avatar"
            src="http://mockavatarurl.com"
            width="50px"
/>,
          <div
            style={
                        Object {
                                    "marginLeft": "2rem",
                                  }
            }
>
            <h3>
                         
                        george
            </h3>
            <span>
                        Contributions: 
                        4
            </span>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "user avatar",
            "src": "http://mockavatarurl.com",
            "width": "50px",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                 
                george
</h3>,
              <span>
                Contributions: 
                4
</span>,
            ],
            "style": Object {
              "marginLeft": "2rem",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "george",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "george",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Contributions: ",
                  4,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Contributions: ",
                4,
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
